src/libxtp/qmmolecule.cc

```diff
@@ -16,33 +16,58 @@
  *
  */
 
-#include <votca/csg/pdbwriter.h>
-#include <votca/csg/xyzreader.h>
-#include <votca/csg/xyzwriter.h>
+#include <votca/csg/io/pdbwriter.h>
+#include <votca/csg/io/xyzreader.h>
+#include <votca/csg/io/xyzwriter.h>
+#include <votca/csg/csgtopology.h>
 #include <votca/tools/elements.h>
 #include <votca/xtp/checkpointreader.h>
 #include <votca/xtp/checkpointwriter.h>
 #include <votca/xtp/qmmolecule.h>
+#include "../../include/votca/xtp/fileadapter.h"
 
 using namespace std;
 using namespace votca::tools;
 
 namespace votca {
 namespace xtp {
 
+const tools::DistanceUnit QMMolecule::distance_unit =                             
+    tools::DistanceUnit::angstroms;                                             
+const tools::MassUnit QMMolecule::mass_unit =                                     
+    tools::MassUnit::atomic_mass_units;                                         
+const tools::TimeUnit QMMolecule::time_unit = tools::TimeUnit::seconds;            
+const tools::ChargeUnit QMMolecule::charge_unit = tools::ChargeUnit::e;           
+const tools::EnergyUnit QMMolecule::energy_unit =                                 
+    tools::EnergyUnit::hartrees; 
+
+  tools::StructureParameters QMMolecule::getParameters() const {
+    tools::StructureParameters params;
+    params.set(tools::StructureParameter::AtomContainerType,getName());
+    params.set(tools::StructureParameter::AtomContainerId,getId());
+    return params;
+  }
+
 void QMMolecule::WriteXYZ(std::string filename, std::string header) const {
-  csg::XYZWriter writer;
+  csg::XYZWriter<csg::CSG_Topology> writer;
   writer.Open(filename, false);
-  writer.Write(*this, header);
+  writer.WriteHeader(header,size());
+  FileAdapter file_adapter;
+  csg::CSG_Topology csg_temp;
+  file_adapter.convertXTPContainerToCSGTopology(*this,csg_temp);
+  writer.Write(csg_temp);
   writer.Close();
   return;
 }
 
 void QMMolecule::LoadFromFile(std::string filename) {
-  csg::XYZReader reader;
-  reader.Open(filename);
-  reader.ReadFile<QMMolecule>(*this);
-  reader.Close();
+  csg::XYZReader<csg::CSG_Topology> reader;
+  //reader.Open(filename);
+  csg::CSG_Topology csg_temp;
+  reader.ReadTopology(filename,csg_temp);
```

Comment:
I do not like the temp stuff

Reply:
I had it before but better is *this=bla.readfile(filename=

Reply:
Ok Will see about changing.

Reply:
Ok, this would prove a real headache to change, because currently these methods return a bool which is used in several parts of csg and elsewhere. At the moment the cost of changing this is high and the reward is low. We can create an issue if it really bugs you but it's probably not worth pursuing.